models/palette_model.py

```diff
@@ -275,6 +275,12 @@ def __init__(self, opt, rank):
             self.opt.f_s_semantic_nclasses, self.opt.cls_semantic_nclasses
         )
 
+        self.use_ref = (
+            self.opt.alg_palette_cond_image_creation == "ref"
```

Comment:
This check should be on the dataloader instead no ? Since there can't be ref data without using the proper dataloader.